kernelci/data/kernelci_api.py

```diff
@@ -94,6 +103,45 @@ def get_root_node(self, node_id):
         resp = self._get('/'.join(['get_root_node', node_id]))
         return json.loads(resp.text)
 
+    def pubsub_event_filter(self, sub_id, event):
+        """Filter Pub/Sub events
+
+        Filter received Pub/Sub event using provided filter dictionary.
+        Return True if the event matches with the filter, otherwise False.
+        """
+        event_filter_status = True
+        filters = self._filters.get(sub_id)
+        if not filters:
+            return event_filter_status
+        for key, value in filters.items():
+            if key not in event.keys():
+                continue
+            if isinstance(value, dict):
+                for sub_key, sub_value in value.items():
+                    if sub_key not in event.get(key):
+                        continue
+                    if sub_value != event.get(key).get(sub_key):
+                        return
```

Comment:
The docstring says the returned value should be a boolean, but here's returning `None`. This will evaluate to false with a basic `if` test but not if the returned value is compared with `False`. So it might work with some luck on how the returned value is used.
```suggestion
                        return False
```